src/coredata.cs

```diff
@@ -1140,23 +1312,147 @@ public interface NSPersistentStore {
 
 		[Export ("didAddToPersistentStoreCoordinator:")]
 		void DidAddToPersistentStoreCoordinator (NSPersistentStoreCoordinator coordinator);
 
 		[Export ("willRemoveFromPersistentStoreCoordinator:")]
-		void WillRemoveFromPersistentStoreCoordinator (NSPersistentStoreCoordinator coordinator);
+		void WillRemoveFromPersistentStoreCoordinator ([NullAllowed] NSPersistentStoreCoordinator coordinator);
 
 		[Since (5,0)]
 		[Field ("NSPersistentStoreSaveConflictsErrorKey")]
 		NSString SaveConflictsErrorKey { get; }
 
 	}
+	
+	[Watch (3,0), TV (10,0), iOS (10,0)]
+	[BaseType (typeof(NSObject))]
+	public interface NSPersistentStoreDescription : NSCopying
+	{
+		// +(instancetype _Nonnull)persistentStoreDescriptionWithURL:(NSURL * _Nonnull)URL;
+		[Static]
+		[Export ("persistentStoreDescriptionWithURL:")]
+		NSPersistentStoreDescription PersistentStoreDescriptionWithUrl (NSUrl Url);
+
+		// @property (copy) NSString * _Nonnull type;
+		[Export ("type")]
+		string Type { get; set; }
+
+		// @property (copy) NSString * _Nullable configuration;
+		[NullAllowed, Export ("configuration")]
+		string Configuration { get; set; }
+
+		// @property (copy) NSURL * _Nullable URL;
+		[NullAllowed, Export ("URL", ArgumentSemantic.Copy)]
+		NSUrl Url { get; set; }
+
+		// @property (readonly, copy, nonatomic) NSDictionary<NSString *,NSObject *> * _Nonnull options;
+		[Export ("options", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSObject> Options { get; }
+
+		// -(void)setOption:(NSObject * _Nullable)option forKey:(NSString * _Nonnull)key;
+		[Export ("setOption:forKey:")]
+		void SetOption ([NullAllowed] NSObject option, string key);
+
+		// @property (getter = isReadOnly) BOOL readOnly;
+		[Export ("readOnly")]
+		bool ReadOnly { [Bind ("isReadOnly")] get; set; }
+
+		// @property NSTimeInterval timeout;
+		[Export ("timeout")]
+		double Timeout { get; set; }
+
+		// @property (readonly, copy, nonatomic) NSDictionary<NSString *,NSObject *> * _Nonnull sqlitePragmas;
+		[Export ("sqlitePragmas", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSObject> SqlitePragmas { get; }
+
+		// -(void)setValue:(NSObject * _Nullable)value forPragmaNamed:(NSString * _Nonnull)name;
+		[Export ("setValue:forPragmaNamed:")]
+		void SetValue ([NullAllowed] NSObject value, string name);
+
+		// @property BOOL shouldAddStoreAsynchronously;
+		[Export ("shouldAddStoreAsynchronously")]
+		bool ShouldAddStoreAsynchronously { get; set; }
+
+		// @property BOOL shouldMigrateStoreAutomatically;
+		[Export ("shouldMigrateStoreAutomatically")]
+		bool ShouldMigrateStoreAutomatically { get; set; }
+
+		// @property BOOL shouldInferMappingModelAutomatically;
+		[Export ("shouldInferMappingModelAutomatically")]
+		bool ShouldInferMappingModelAutomatically { get; set; }
+
+		// -(instancetype _Nonnull)initWithURL:(NSURL * _Nonnull)url __attribute__((objc_designated_initializer));
+		[Export ("initWithURL:")]
+		[DesignatedInitializer]
```

Comment:
let's add `[DisableDefaultCtor]` on the type since there's a designated initializer, so it's likely not usable (and easier to add, than remove, later if it's allowed)